bundles/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/internal/handler/PHCBridgeHandler.java

```diff
@@ -104,215 +110,405 @@ public void initialize() {
+
+                @Override
+                public void run() {
+                    processSendQueue();
+                }
 
+            }.start();
         } catch (PortInUseException | TooManyListenersException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Could not open serial port " + serialPort + ": " + e.getMessage());
-
+                    "Could not open serial port " + port + ": " + e.getMessage());
         } catch (UnsupportedCommOperationException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Could not configure serial port " + serialPort + ": " + e.getMessage());
-        } catch (NoSuchPortException e) {
-            StringBuilder sb = new StringBuilder();
-            @SuppressWarnings("unchecked")
-            Enumeration<CommPortIdentifier> portList = CommPortIdentifier.getPortIdentifiers();
-            while (portList.hasMoreElements()) {
-                CommPortIdentifier id = portList.nextElement();
-                if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
-                    sb.append(id.getName() + "\n");
-                }
-            }
-            logger.warn("Serial port '{}' could not be found. Available ports are:\n {}", port, sb);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Serial port '" + port + "' could not be found.");
+                    "Could not configure serial port " + port + ": " + e.getMessage());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Failed to get input or output stream for serialPort. (See debugging log.)");
+            logger.debug("Failed to get inputstream for serialPort", e);
         }
     }
 
     /**
-     * Puts the available data on serial port into a buffer and transfer it to the processing method.
+     * Reads the data on serial port and puts it into the internal buffer.
      */
     @Override
     public void serialEvent(SerialPortEvent event) {
-        if (event.getEventType() != SerialPortEvent.DATA_AVAILABLE) {
-            return;
+        if (event.getEventType() == SerialPortEvent.DATA_AVAILABLE && serialIn != null) {
+            try {
+                byte[] bytes = new byte[serialIn.available()];
+                serialIn.read(bytes);
+
+                buffer.offer(bytes);
+                if (logger.isTraceEnabled()) {
+                    StringBuilder hex = new StringBuilder();
+                    for (byte b : bytes) {
+                        hex.append(PHCHelper.byteToHexString(b));
+                    }
+                    logger.trace("buffer offered {}", hex);
+                }
+            } catch (IOException e) {
+                logger.error("error on reading input stream to internal buffer", e);
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.

Also check other occurrences.

And consider whether your callers should be notified ot the error, now there is zero feedback.